app/models/host/base.rb

```diff
@@ -214,17 +214,20 @@ def set_comment(parser)
       desc 'Note that available facts depend on what facts have been uploaded to Foreman,
            typical sources are Puppet facter, subscription manager etc.
            The facts can be out of date, this macro only provides access to the value stored in the database.'
+      optional :fact_names, Array, desc: 'A list of fact names to return. If empty all facts are returned'
       returns Hash, desc: 'A hash of facts, keys are fact names, values are fact values'
       example '@host.facts # => { "hardwareisa"=>"x86_64", "kernel"=>"Linux", "virtual"=>"physical", ... }', desc: 'Getting all host facts'
       example '@host.facts["uptime"] # => "30 days"', desc: 'Getting specific fact value, +uptime+ in this case'
       aliases :facts
     end
-    def facts_hash
-      hash = {}
-      fact_values.includes(:fact_name).collect do |fact|
-        hash[fact.fact_name.name] = fact.value
-      end
-      hash
+    def facts_hash(*fact_names)
+      # keep it to one SQL query
+      query = if fact_names.present?
```

Comment:
I'm much more used to plain Ruby where `Array.present?` doesn't exist. Would it make sense to stick to plain Ruby and use `.any?`:
```suggestion
      query = if fact_names.any?
```